### PR TITLE
AZ_EXPECT_SUCCESS

### PR DESCRIPTION
**Pull request by antkmsft**
AZ_EXPECT_SUCCESS is better than ignore_result; Use it for cases when  writing a test consisting of complex steps, and things you use for preparations return az_spans, but yor test is not to test them. Use AZ_TEST_AZZERT(result == AZ_OK) when verifying the function that you are testing; but for all other functions returning az_result, use AZ_EXPECT_SUCCESS, so that your code is up to 2x shorter, you don't step on AZ_NODISCARD every other line, and the result is not ignored completely + it is easy to identify all the places where you use them. Plus it is only defined for tests.

### CODE REVIEW

File: sdk/core/core/test/az_test.h [hunk 1]
```diff
@@ -13,3 +15,9 @@ extern int exit_code;
       exit_code = 1; \
     } \
   } while (false);
+
+#define AZ_EXPECT_SUCCESS(exp) \
```
Comment (sergey-shandar) [inline sdk/core/core/test/az_test.h]: `#define AZ_EXPECT_SUCCESS(exp) TEST_ASSERT(!az_failed(exp))`

File: sdk/core/core/test/az_test.h [hunk 2]
```diff
@@ -13,3 +15,5 @@ extern int exit_code;
       exit_code = 1; \
     } \
   } while (false);
+
+#define AZ_EXPECT_SUCCESS(exp) TEST_ASSERT(!az_failed(_result))
```
Comment (antkmsft) [inline sdk/core/core/test/az_test.h]:
```suggestion
#define AZ_EXPECT_SUCCESS(exp) TEST_ASSERT(!az_failed(exp))
```

File: sdk/core/core/test/az_test.h [hunk 3]
```diff
@@ -13,3 +15,5 @@ extern int exit_code;
       exit_code = 1; \
     } \
   } while (false);
+
+#define TEST_EXPECT_SUCCESS(exp) TEST_ASSERT(!az_failed(exp))
```
Comment (vhvb1989) [inline sdk/core/core/test/az_test.h]: az_succeeded() would be better than !az_failed()

**Comment by antkmsft** [comment]
Will go in as part of https://github.com/Azure/azure-sdk-for-c/pull/175

File: sdk/core/core/test/az_test.h (hunk 3 above)
Comment (antkmsft) [reply sdk/core/core/test/az_test.h]: Fixed.

**Pull request closed without merging**